### DISCUSSION TITLE
Add Flow.onErrorComplete operator

### DISCUSSION THREAD

**Discussion by He-Pin** [apache/pekko#623] [General]
When we encounter an exception, sometime we may don't want the flow to result in an error but completes normally.

The implementation can be simple, but still cause some confusing.

I would like suggest to add a `onErrorComplete` operator which will complete the flow on some dedicated condition.

And better we have another `onErrorContiune` operator, which will keep the flow working but will need the operator before it to support this.

**Comment by pjfanning** [comment]
Moved to https://github.com/apache/incubator-pekko/issues/624 and otherwise, there is little point separating this discussion from #605
